src/main/java/org/opensrp/queue/RabbitMQReceiver.java

```diff
@@ -0,0 +1,97 @@
+package org.opensrp.queue;
+
+import com.ibm.fhir.model.format.Format;
+import com.ibm.fhir.model.parser.FHIRParser;
+import com.ibm.fhir.model.parser.exception.FHIRParserException;
+import com.ibm.fhir.model.resource.DomainResource;
+import org.opensrp.repository.ClientsRepository;
+import org.opensrp.repository.EventsRepository;
+import org.opensrp.repository.LocationRepository;
+import org.opensrp.repository.TaskRepository;
+import org.opensrp.service.PlanService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.smartregister.domain.PlanDefinition;
+import org.smartregister.pathevaluator.PathEvaluatorLibrary;
+import org.smartregister.pathevaluator.plan.PlanEvaluator;
+import org.springframework.amqp.rabbit.annotation.RabbitHandler;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+@Component
+@RabbitListener(queues = "rabbitmq.task.queue", id = "listener")
+public class RabbitMQReceiver {
+
+	private PlanEvaluator planEvaluator;
+
+	@Autowired
+	private LocationRepository locationRepository;
+
+	@Autowired
+	private ClientsRepository clientsRepository;
+
+	@Autowired
+	private TaskRepository taskRepository;
+
+	@Autowired
+	private EventsRepository eventsRepository;
+
+	@Autowired
+	private PlanService planService;
+
+	private FHIRParser fhirParser;
+
+	private static Logger logger = LoggerFactory.getLogger(RabbitMQReceiver.class.toString());
+
+	@PostConstruct
+	public void init() {
+		PathEvaluatorLibrary.init(locationRepository, clientsRepository, taskRepository, eventsRepository);
+		planEvaluator = new PlanEvaluator("");
```

Comment:
This should be created for each plan being processed, we would want to track the user activating the plan. Pass this as a param from the plan evaluator library

Reply:
Added parameter of the username in queueHelper and passing it from there into PlanEvaluator. Updated, Please review and provide your feedback @githengi